### DISCUSSION TITLE
Ktor subscription example (?)

### DISCUSSION THREAD

**Discussion by Mautriz** [ExpediaGroup/graphql-kotlin#1210] [Q&A]
Hey there, can someone help me understanding how to implement subscriptions with websockets on ktor or make a small example? I wanted to use kotlin/graphql in a new project and subscriptions are whats blocking me (I'm new to kotlin btw)

**Comment by smyrick** [comment]
We do not have an example in Ktor for subscriptions as there are many implementation details that are specific for each server. If you take a look at the [Spring code](https://github.com/ExpediaGroup/graphql-kotlin/tree/master/servers/graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/server/spring/subscriptions) you will see we are implementing the [subscriptions-transport-ws protocol](https://github.com/apollographql/subscriptions-transport-ws)


You could also implement the [graphql-ws protocol](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md)

**Comment by robertoaltieri** [comment]
@Mautriz did you find how to use subscriptions? @smyrick you still don't have an example for when `graphql-ws` protocol is used?
